subprojects/core/src/main/java/org/gradle/api/internal/initialization/transform/BaseInstrumentingArtifactTransform.java

```diff
@@ -79,53 +76,30 @@ public interface Parameters extends TransformParameters {
     @InputArtifact
     public abstract Provider<FileSystemLocation> getInput();
 
-    protected void execute(@Nullable File artifactToInstrument, TransformOutputs outputs, Consumer<TransformOutputs> originalFileProducer) {
+    protected void doTransform(File artifactToTransform, TransformOutputs outputs) {
         createInstrumentationClasspathMarker(outputs);
-        if (artifactToInstrument == null || !artifactToInstrument.exists()) {
-            createNewFile(outputs.file(ORIGINAL_FILE_DOES_NOT_EXIST_MARKER));
+        if (!artifactToTransform.exists()) {
+            GFileUtils.touch(outputs.file(ORIGINAL_FILE_DOES_NOT_EXIST_MARKER));
```

Comment:
❓ Why this change?

Reply:
In previous iteration `artifactToInstrument` passed here could be null, but with the change that should not be possible anymore, since we filter unexisting jars out already before that step.

Reply:
Can we then keep using `createNewFile()`? Using `touch` gives the idea that we expect outputs to sometimes exist here.